agent/bench-scripts/tests/pbench-fio/test-04.opts

```diff
@@ -1 +1 @@
---test-types=rw,randrw
+--test-types=rw,randrw --ramptime=61
```

Comment:
The job file for `test-04` provides `runtime=30`, which I assume is 30 seconds; and, it (with this change) provides `ramp_time=61`.  Am I correct in assuming that the "runtime" doesn't start until _after_ the "ramp_time" expires?  (I.e., I just want to confirm that the test won't end before it starts....)

Reply:
Oh I thought the tests are mocked and just randomly selected one and added this argument to have it covered. In this case let me chose 1s ramptime, which should actually decrease the duration.

Anyway, yes, the ramp_time argument should result in fio starting the workload, waiting for `ramp_time` and starting the measurement when it runs out of that time.

Reply:
The legacy unit test environment is ... weird. It's a partial functional test, but with a `PATH` that overrides a lot of commands, including, relevant to `fio`, `screen` and `ssh`. Since we run `fio` via `ssh` commands (or local `screen` commands in PR #2542) all of the "real work" is essentially mocked out by a `test-bin` script that just echoes the command line into a log. But there is also mocking of the postprocessing, and so what actually happens in the "tool pipeline" is a bit more complicated than that, and often dependent on the environments configured for individual tests.

The best way to run the legacy unit tests (and the actual unit pytests) from a git checkout branch is to mimic our Jenkins run as found in `jenkins/Pipeline.gy`:

```
jenkins/run "source jenkins/python-setup.sh && jenkins/run-pytests && jenkins/run-unittests"
```

You can do the `jenkins/run-pytests` separately from `jenkins/run-unittests`, just remember to include the `source` command first do to the environment setup in the container fired up from `jenkins/run`.